### ISSUE TITLE
Images: Add alt attributes to the images inserted, and in the demo content

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#1309]
In order to show best practices and also for proper testing, I'd propose to add alt attributes to the images in the demo content. At least empty alt attributes `alt=""` 🙂 

Worth noting images don't inherit the alt text or the caption from the media library, I think they should. Not sure if this has already been discussed or addressed in a separate issue. /cc @jasmussen

**Comment by jasmussen** [comment]
So it seems that not only do images in post content not have alt text, images inserted from the library don't either :|

I'm going to update this ticket label to reflect that images inserted should also have that.

**Comment by afercia** [comment]
@jasmussen thanks.
One interesting thing we've recently discussed in the accessibility team, during the review of the new media widgets, is the different purpose the alt text has in the front-end and in an editing context (ie. when editing a post).

As a general rule, we all know that (I'll simplify a bit):
- purely decorative images should have an empty alt text `alt=""`
- images that convey relevant information, should use a meaningful alt text

However, when in an editing context, things are slightly different. Will try to do my best to explain. While editing a post, users need to know what the image they've added is. 
When users add an alt text to the image because they want to communicate the image purpose in the front-end, that also serves the purpose to make the image recognizable in the editor.
However, when the image is purely decorative, users will set an empty alt text. That's perfectly correct for the front-end but it will also make the image not recognizable for screen reader users because images with an `alt=""` are just ignored.

Ideally, when in the editor, also images intended to use an empty `alt=""` should have _something_ to make them perceivable and announced by screen readers. Not sure about the best option, but maybe this is something worth considering as a nice improvement Gutenberg could bring in.
/cc @joedolson

As a reference, the solution implemented for the image widget was to remove the empty alt attribute in the editor when there's no alt value, so at least the filename gets announced. Not sure that's the best available option and maybe we could consider an improved solution.
https://github.com/xwp/wp-core-media-widgets/issues/47
https://github.com/xwp/wp-core-media-widgets/pull/50

**Comment by joedolson** [comment]
@afercia Do you want to open a new issue per your comment about alt attributes in an editor context or re-open this issue? I think it's a separate issue, and should be probably be raised separately, so that it can be discussed without needing to hold this open.

**Comment by afercia** [comment]
@joedolson done in #1520
